### PR TITLE
Issue 1/convert mock mvc tests to spring boot

### LINKED ISSUE #1: Convert on of the mockMvc tests to springBoot restTemplate one
Convert BusinessControllerE2E to one using @SpringBootTest and move it to modules folder following principles of other classes. 

See BusinessControllerE2E for example and refactor test contents in BaseClass if duplications occur.

### PR DESCRIPTION
**Pull request by MBarzyk**
- created a base class with MariaDB config which can be reused in every integration/E2E class
- dropped mockMvc for BusinessControllerE2E and started using TestRestTemplate from spring boot

### CODE REVIEW

File: stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    static public void provideProperties(DynamicPropertyRegistry dynamicPropertyRegistry) {
+        dynamicPropertyRegistry.add("db.mysql.db", () -> mariadb.getDatabaseName());
+        dynamicPropertyRegistry.add("db.mysql.host", () -> mariadb.getHost());
+        dynamicPropertyRegistry.add("db.mysql.port", () -> mariadb.getFirstMappedPort());
+        dynamicPropertyRegistry.add("db.mysql.username", () -> mariadb.getUsername());
+        dynamicPropertyRegistry.add("db.mysql.password", () -> mariadb.getPassword());
+    }
+
+    @BeforeAll
+    static public void startMariaDBWithLoggingForClass(Class loggedClass) {
```
Comment (oneacik) [inline stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java:30]: does it work?
How does it find loggedClass, when it is marked as BeforeAll?

File: stamply-backend/src/test/java/com/example/controller/business/BusinessControllerE2E.java [hunk 2]
```diff
@@ -45,47 +46,50 @@ public class BusinessControllerE2E extends AbstractDatabaseTest {
...
 
     @Test
-    public void givenNonExistingBusinessWhenGotThenReturns404() throws Exception {
-        final var mvcResult = mockMvc.perform(
-                get("/api/business/")
-                        .accept(MediaType.APPLICATION_JSON)
-                        .contentType(MediaType.APPLICATION_JSON)
-                        .header("Authorization", "Bearer " + createToken())
-        ).andExpect(status().is(404)).andReturn();
+    public void givenNonExistingBusinessWhenGotThenReturns404() {
```
Comment (oneacik) [inline stamply-backend/src/test/java/com/example/controller/business/BusinessControllerE2E.java:65]: does it work when ran by intellijIdea?
if it doesn't extend BaseSpringBootIT, then provideProperties won't be able to provide properties for mariaDB, therefore it won't be used.
Maven then is providing database.

File: stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java (hunk 1 above)
Comment (oneacik) [reply stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java:30]: You should probaly remove BeforeAll if you want to use this method in tests with Class to not mislead.

Comment (MBarzyk) [reply stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java:30]: Removed `BeforeAll` but from the super class, it was unnecessary there, and now the test runs smoothly using `mvn clean verify`

Comment (MBarzyk) [reply stamply-backend/src/test/java/com/example/common/BaseSpringBootIT.java:30]: Ye, runs smoothly after removing the annotation from super class <3

**Review by oneacik — APPROVED** [review]

**Pull request merged**